[jdbc] Storage of ImageItem broken

If I use an Image item and want to store its value to a jdbc database, only meta information is stored.
This is because `ImageItem.getState.toString` produces only meta information of an image.
Example:

items:
`Image Doorbell_Image "Doorbell Image [%s]" (persistGroup) { channel="doorbird:d101:doorbell:image" }
`

persistence:
```
Strategies {
        // if no strategy is specified for an item entry below, the default list will be used
        everyMinute     : "0 * * * * ?"
        every5Minutes : "0 */5 * * * ?"
        everyHour   : "0 0 * * * ?"
        everyDay    : "0 0 0 * * ?"
        default = everyChange
}

Items {
    persistGroup* : strategy = everyChange, everyDay, restoreOnStartup
}
```

This produces following entry in a database table:

```
mysql> select * from Item384;
+---------------------+------------------------------------+
| Time                | Value                              |
+---------------------+------------------------------------+
| 2021-08-19 15:11:29 | raw type (image/jpeg): 55033 bytes |
+---------------------+------------------------------------+

```

The jdbc logging looks as follows:

```
2021-08-22 20:37:16.975 [DEBUG] [persistence.jdbc.internal.JdbcMapper] - JDBC::storeItemValue: item=Doorbell_Image (Type=ImageItem, State=raw type (image/jpeg): 54256 bytes, Label=Doorbell Image, Category=null, Groups=[persistGroup])
2021-08-22 20:37:16.975 [DEBUG] [.openhab.persistence.jdbc.dto.ItemVO] - JDBC:ItemVO tableName=Item384; newTableName=null;
2021-08-22 20:37:16.975 [DEBUG] [nhab.persistence.jdbc.db.JdbcBaseDAO] - JDBC::getItemType: Try to use ItemType IMAGEITEM for Item Doorbell_Image
2021-08-22 20:37:16.975 [DEBUG] [nhab.persistence.jdbc.db.JdbcBaseDAO] - JDBC::storeItemValueProvider: item 'Doorbell_Image' as Type 'IMAGEITEM' in 'Item384' with state 'raw type (image/jpeg): 54256 bytes'
2021-08-22 20:37:16.975 [DEBUG] [nhab.persistence.jdbc.db.JdbcBaseDAO] - JDBC::storeItemValueProvider: getState: 'raw type (image/jpeg): 54256 bytes'
2021-08-22 20:37:16.976 [DEBUG] [.openhab.persistence.jdbc.dto.ItemVO] - JDBC:ItemVO setValueTypes dbType=BLOB; javaType=class java.lang.String;
2021-08-22 20:37:16.976 [DEBUG] [nhab.persistence.jdbc.db.JdbcBaseDAO] - JDBC::storeItemValueProvider: other: item.getState().toString(): 'raw type (image/jpeg): 54256 bytes'
2021-08-22 20:37:16.976 [DEBUG] [nhab.persistence.jdbc.db.JdbcBaseDAO] - JDBC::doStoreItemValue sql=INSERT INTO Item384 (TIME, VALUE) VALUES( NOW(3), ? ) ON DUPLICATE KEY UPDATE VALUE= ? value='raw type (image/jpeg): 54256 bytes'
2021-08-22 20:37:16.979 [DEBUG] [jdbc.internal.JdbcPersistenceService] - JDBC: Stored item 'Doorbell_Image' as 'raw type (image/jpeg): 54256 bytes' in SQL database at Sun Aug 22 20:37:16 CEST 2021 in 3 ms.

```

On restore this produces the following exception, because the Image meta info stored cannot restore in the item state:

```
2021-08-19 15:17:24.768 [ERROR] [nal.common.AbstractInvocationHandler] - An error occurred while calling method 'Queryab
lePersistenceService.query()' on 'org.openhab.persistence.jdbc.internal.JdbcPersistenceService@5537295e': Invalid data U
RI syntax for argument raw type (image/jpeg): 54554 bytes
java.lang.IllegalArgumentException: Invalid data URI syntax for argument raw type (image/jpeg): 54554 bytes
        at org.openhab.core.library.types.RawType.valueOf(RawType.java:59) ~[bundleFile:?]
        at org.openhab.persistence.jdbc.db.JdbcBaseDAO.getState(JdbcBaseDAO.java:512) ~[?:?]
        at org.openhab.persistence.jdbc.db.JdbcBaseDAO.lambda$0(JdbcBaseDAO.java:344) ~[?:?]
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195) ~[?:?]
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655) ~[?:?]
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484) ~[?:?]
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474) ~[?:?]
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913) ~[?:?]
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578) ~[?:?]
        at org.openhab.persistence.jdbc.db.JdbcBaseDAO.doGetHistItemFilterQuery(JdbcBaseDAO.java:345) ~[?:?]
        at org.openhab.persistence.jdbc.internal.JdbcMapper.getHistItemFilterQuery(JdbcMapper.java:169) ~[?:?]
        at org.openhab.persistence.jdbc.internal.JdbcPersistenceService.query(JdbcPersistenceService.java:205) ~[?:?]
        at jdk.internal.reflect.GeneratedMethodAccessor113.invoke(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.openhab.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:154) [bundleFile:?]
        at org.openhab.core.internal.common.Invocation.call(Invocation.java:52) [bundleFile:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:829) [?:?]

```

## Expected Behavior

The image value (base64 encoded image) should be stored instead of the meta information.

## Current Behavior

Currently the meta information about the image is stored.

## Possible Solution

I guess the method `storeItemValueProvider` in `JDBCBaseDAO` (https://github.com/openhab/openhab-addons/blob/73f7a86dd827bc93d6b78cc1bdaffb0159d80de5/bundles/org.openhab.persistence.jdbc/src/main/java/org/openhab/persistence/jdbc/db/JdbcBaseDAO.java#L397) needs to be adapted and a special handling for image items needs to be added:

```
case "IMAGEITEM":
                vo.setValueTypes(getSqlTypes().get(itemType), java.lang.String.class);
			    String sValue = item.getState().toFullString();
                logger.debug("JDBC::storeItemValueProvider: ImageItem: '{}'", sValue);
                vo.setValue(sValue);
                break;

```

## Steps to Reproduce (for Bugs)
see above 

## Context
I want to store Images of my doorbell camera (only those if someone rings the bell).

## Your Environment
openHAB 3.1.0 (stable)
Persistence Binding 3.1.0
OpenJDK 64-Bit Server VM (build 11.0.11+9-Ubuntu-0ubuntu2.18.04, mixed mode, sharing)

--- a/bundles/org.openhab.persistence.jdbc/src/main/java/org/openhab/persistence/jdbc/db/JdbcBaseDAO.java
+++ b/bundles/org.openhab.persistence.jdbc/src/main/java/org/openhab/persistence/jdbc/db/JdbcBaseDAO.java
@@ -480,6 +480,12 @@ public class JdbcBaseDAO {
                 logger.debug("JDBC::storeItemValueProvider: DateTimeItem: '{}'", d);
                 vo.setValue(d);
                 break;
+            case "IMAGEITEM":
+                vo.setValueTypes(getSqlTypes().get(itemType), java.lang.String.class);
+                String encodedString = item.getState().toFullString();
+                logger.debug("JDBC::storeItemValueProvider: ImageItem: '{}'", encodedString);
+                vo.setValue(encodedString);
+                break;
             default:
                 // All other items should return the best format by default
                 vo.setValueTypes(getSqlTypes().get(itemType), java.lang.String.class);

